content/en/tutorials/using-terraform-with-testcontainers-and-localstack/index.md

```diff
@@ -0,0 +1,372 @@
+
+```bash
+$ export LOCALSTACK_AUTH_TOKEN=<your_auth_token>
+$ localstack start -e EXTENSION_AUTO_INSTALL=localstack-extension-terraform-init 
+                -v ./terraform/main.tf:/etc/localstack/init/ready.d/main.tf 
+                -v ./target/product-lambda.jar:/etc/localstack/init/ready.d/target/product-lambda.jar
+```
+
+This is the easiest way to quickly spin up the desired services at startup.
+The command starts LocalStack with the configuration to automatically install the **`localstack-extension-terraform-init`** [extension]({{<ref "/user-guide/extensions/">}}) and
+mount the necessary files into the container: the Terraform configuration file and the Lambda JAR file.
+The extension will install both `terraform` and `tflocal` into your LocalStack container, and enable the init hook runners to detect Terraform files.
+You can also organize your Terraform files into subdirectories if you want.
+
+If Docker gives you any trouble regarding mounting permissions, you can add `$(pwd)` in front of the local paths, to turn them into
+absolute paths.
+
+Since the initialization hook runs `terraform init`, the AWS Terraform provider will be downloaded in the container every time.
+You can avoid this by mounting a directory instead of a single file.
+Any Terraform state including the `.terraform` folder that contains the provider, will be cached on your host directory, however they may require `sudo` permissions to modify or delete, as they are created by the container.
+
+
+#### Docker compose
+
+Another way of starting LocalStack with the desired services is using `docker compose`. In the root folder, you'll find the essential configs in the `docker-compose.yml` file:
+<details>
+<summary><b>Expand file</b></summary>
+{{< command >}}
+version: "3.8"
+
+services:
+    localstack:
+        container_name: "${LOCALSTACK_DOCKER_NAME:-localstack-main}"
+        image: localstack/localstack-pro:latest  # required for Pro
+        ports:
+            - "127.0.0.1:4566:4566"            # LocalStack Gateway
+            - "127.0.0.1:4510-4559:4510-4559"  # external services port range
+            - "127.0.0.1:443:443"              # LocalStack HTTPS Gateway (Pro)
+        environment:
+            # Activate LocalStack Pro: https://docs.localstack.cloud/getting-started/auth-token/
+            - LOCALSTACK_AUTH_TOKEN=${LOCALSTACK_AUTH_TOKEN:?}  # required for Pro
+            # LocalStack configuration: https://docs.localstack.cloud/references/configuration/
+            - DEBUG=1
+            - PERSISTENCE=${PERSISTENCE:-0}
+            - EXTENSION_AUTO_INSTALL=localstack-extension-terraform-init
+        volumes:
+            - "${LOCALSTACK_VOLUME_DIR:-./volume}:/var/lib/localstack"
+            - "/var/run/docker.sock:/var/run/docker.sock"
+            - "./terraform:/etc/localstack/init/ready.d"
+            - "./target/product-lambda.jar:/etc/localstack/init/ready.d/target/product-lambda.jar"
+{{< / command >}}
```

Comment:
Not exactly a command — use code formatting to leverage YAML syntax highlighting